src/mono/mono/mini/simd-intrinsics.c

```diff
@@ -3093,16 +3209,113 @@ emit_vector_2_3_4 (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSignature *f
 
 		return min;
 	}
-	case SN_Conjugate:
-	case SN_Distance:
-	case SN_DistanceSquared:
+	case SN_Distance: 
+	case SN_DistanceSquared: {
+#if defined(TARGET_ARM64)
+		MonoInst *diffs = emit_simd_ins_for_sig (cfg, klass, OP_XBINOP, OP_FSUB, MONO_TYPE_R4, fsig, args);
+		MonoInst *diffs_squared = emit_simd_ins (cfg, klass, OP_XBINOP, diffs->dreg, diffs->dreg);
+		diffs_squared->inst_c0 = OP_FMUL;
+		diffs_squared->inst_c1 = MONO_TYPE_R4;
+
+		switch (id) {
```

Comment:
Nit. maybe we could also use dot product code here and optionally use sqrt depending on the method called.

Reply:
👍. This is really just `Dot(value1 - value2)` and then optionally `sqrt`

`Dot` is a fairly core and frequently reused function (which itself is either `DPPS` or `MUL+SUM`) and so it likely warrants a helper more than `sum_sqrt`